File: android/lib/src/main/java/org/elastos/sdk/elephantwallet/contact/internal/IdentifyCode.java
package org.elastos.sdk.elephantwallet.contact.internal;

import android.support.annotation.CallSuper;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.annotations.SerializedName;

import java.util.HashMap;
import java.util.List;

public class IdentifyCode implements IdentifyCodeInterface {
    @SerializedName(JsonKey.IdCodeMap)
    protected HashMap<Integer, String> idCodeMap;
    @SerializedName(JsonKey.UpdateTime)
    public long updateTime;

    public String phoneNumber;
    public String emailAddress;
    public String wechatId;

    protected int fromJsonObj(IdentifyCode json) {
        this.idCodeMap = null;
        this.updateTime = json.updateTime;

        this.phoneNumber = json.phoneNumber;
        this.emailAddress = json.emailAddress;
        this.wechatId = json.wechatId;

        if(json.idCodeMap != null) {
            this.phoneNumber = json.idCodeMap.get(Type.PhoneNumber.id());
            this.emailAddress = json.idCodeMap.get(Type.EmailAddress.id());
            this.wechatId = json.idCodeMap.get(Type.WechatId.id());
        }

        return 0;
    }

    protected int toJsonObj(IdentifyCode json) {
        json.idCodeMap = null;
        json.updateTime = this.updateTime;

        json.phoneNumber = this.phoneNumber;
        json.emailAddress = this.emailAddress;
        json.wechatId = this.wechatId;

        if(this.idCodeMap != null) {
            json.phoneNumber = this.idCodeMap.get(Type.PhoneNumber.id());
            json.emailAddress = this.idCodeMap.get(Type.EmailAddress.id());
            json.wechatId = this.idCodeMap.get(Type.WechatId.id());
        }

        return 0;
    }

    IdentifyCode() {
    }
} // class Factory

File: ios/sdk/swift/internal/IdentifyCode.swift


public class IdentifyCode: Codable {
  enum Kind: Int {
       case PhoneNumber = 1
       case EmailAddress = 2
       case WechatId = 3
  }
  
  var idCodeMap: [Int: String]?
  public private(set) var updateTime: Int64 = 0

  public private(set) var phoneNumber: String? = nil
  public private(set) var emailAddress: String? = nil
  public private(set) var wechatId: String? = nil

  enum CodingKeys : String, CodingKey {
    case idCodeMap
    case updateTime
    
    var rawValue: String {
      get {
        switch self {
          case .idCodeMap: return JsonKey.IdCodeMap
          case .updateTime: return JsonKey.UpdateTime
        }
      }
    }
  }

  func fromJsonObj(json: IdentifyCode) -> Int {
    self.idCodeMap = nil
    self.updateTime = json.updateTime

    self.phoneNumber = json.phoneNumber
    self.emailAddress = json.emailAddress
    self.wechatId = json.wechatId

    if json.idCodeMap != nil {
      self.phoneNumber = json.idCodeMap?[Kind.PhoneNumber.rawValue]
      self.emailAddress = json.idCodeMap?[Kind.EmailAddress.rawValue]
      self.wechatId = json.idCodeMap?[Kind.WechatId.rawValue]
    }

    return 0
  }

  func toJsonObj(json: IdentifyCode) -> Int {
    json.idCodeMap = nil
    json.updateTime = self.updateTime

    json.phoneNumber = self.phoneNumber
    json.emailAddress = self.emailAddress
    json.wechatId = self.wechatId

    if self.idCodeMap != nil {
      json.phoneNumber = self.idCodeMap?[Kind.PhoneNumber.rawValue]
      json.emailAddress = self.idCodeMap?[Kind.EmailAddress.rawValue]
      json.wechatId = self.idCodeMap?[Kind.WechatId.rawValue]
    }

    return 0
  }
} // class Factory
